Makefile

```diff
@@ -52,45 +59,53 @@ EMFLAGS_DEBUG = \
 	-O1
 
 BITCODE_FILES = out/sqlite3.bc out/extension-functions.bc
+
 OUTPUT_WRAPPER_FILES = src/shell-pre.js src/shell-post.js
 
+OUTPUT_API_FILES = out/api.js
+
+EMFLAGS_PRE_JS_FILES = \
+	--pre-js out/api.js
+
+EXPORTED_METHODS_JSON_FILES = src/exported_functions.json src/exported_runtime_methods.json
+
 all: optimized debug worker
 
 .PHONY: debug
-debug: dist/sql-asm-debug.js dist/sql-wasm-debug.js
+debug: dist/sql-asm-debug.js dist/sql-wasm-debug.js dist/sql-asm-debug-memory-growth.js
+
+dist/sql-asm-debug.js: $(BITCODE_FILES) $(OUTPUT_WRAPPER_FILES) $(OUTPUT_API_FILES) $(EXPORTED_METHODS_JSON_FILES)
+	$(EMCC) $(EMFLAGS) $(EMFLAGS_DEBUG) $(EMFLAGS_ASM) $(BITCODE_FILES) $(EMFLAGS_PRE_JS_FILES) -o $@
+	cat src/shell-pre.js $@ src/shell-post.js > out/sql-wrapped.js
```

Comment:
I am now having mixed feelings about this change. In case something goes wrong with cat or mv, there could be a chance of having generated dist artifact with the contents of `src/shell-pre.js` and `src/shell-post.js` missing.

I might favor something like this instead (with proper indenting, of course):

    mv $@ out/tmp-sql-raw.js
    cat src/shell-pre.js out/tmp-sql-raw.js src/shell-post.js > $@
    rm out/tmp-sql-raw.js

Maybe it would be easiest if I would just revert ea45e772f256c42f36e886a44d9a61dd9bd05297